lib/src/controls/flyouts/menu.dart

```diff
@@ -0,0 +1,532 @@
+  Widget build(BuildContext context) {
+    return _MenuFlyoutSubItem(key: _key, item: this, items: items);
+  }
+}
+
+class _MenuFlyoutSubItem extends StatefulWidget {
+  final MenuFlyoutSubItem item;
+  final List<MenuFlyoutItemBase> items;
+
+  const _MenuFlyoutSubItem({
+    Key? key,
+    required this.item,
+    required this.items,
+  }) : super(key: key);
+
+  @override
+  State<_MenuFlyoutSubItem> createState() => __MenuFlyoutSubItemState();
+}
+
+class __MenuFlyoutSubItemState extends State<_MenuFlyoutSubItem>
+    with SingleTickerProviderStateMixin {
+  /// The animation controller responsible for the animation of the flyout
+  ///
+  /// The duration is defined at build time
+  late final transitionController = AnimationController(vsync: this);
+
+  final menuKey = GlobalKey<_MenuFlyoutState>();
+
+  Timer? showTimer;
+
+  @override
+  void dispose() {
+    transitionController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final menuInfo = MenuInfoProvider.of(context);
+
+    transitionController.duration = menuInfo.widget.transitionDuration;
+
+    return MouseRegion(
+      onEnter: (event) {
+        showTimer = Timer(widget.item.showHoverDelay, () {
+          show(menuInfo);
+        });
+      },
+      onExit: (event) {
+        if (showTimer != null && showTimer!.isActive) {
+          showTimer!.cancel();
+        }
+      },
+      child: MenuFlyoutItem(
+        key: widget.item.key,
+        text: widget.item.text,
+        leading: widget.item.leading,
+        selected: isShowing(menuInfo),
+        trailing: widget.item.trailing,
+        onPressed: () {
+          if (widget.item.showBehavior != SubItemShowBehavior.press) return;
+
+          show(menuInfo);
+        },
+      ).build(context),
+    );
+  }
+
+  bool isShowing(MenuInfoProviderState menuInfo) {
+    return menuInfo.contains(menuKey);
+  }
+
+  void show(MenuInfoProviderState menuInfo) {
+    final itemBox = context.findRenderObject() as RenderBox;
+    final itemRect = itemBox.localToGlobal(Offset.zero) & itemBox.size;
+
+    menuInfo.add(
+      CustomSingleChildLayout(
+        delegate: _SubItemPositionDelegate(
+          parentRect: itemRect,
+          parentSize: itemBox.size,
+          margin: menuInfo.widget.margin,
+        ),
+        child: ContentManager(
+          content: (_) {
+            return FadeTransition(
+              opacity: transitionController,
+              child: MenuFlyout(
+                key: menuKey,
+                items: widget.items,
+              ),
+            );
+          },
+        ),
+      ),
+      menuKey,
+    );
+
+    transitionController.forward();
+    setState(() {});
+  }
+
+  /// Closes this menu and its children
+  Future<void> close(MenuInfoProviderState menuInfo) async {
+    await Future.wait([
+      if (menuKey.currentState != null)
+        ...menuKey.currentState!.keys
+            .whereType<GlobalKey<__MenuFlyoutSubItemState>>()
```

Comment:
I suggest using a provider that lives on top of the menu to have a cleaner architecture, this can and will break very easily. Use something like a `MenuSubitemRegistry` to keep track of current open subitems